Src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs

```diff
@@ -231,7 +231,7 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
             }, cancellationToken);
             }
 
-            Task.WaitAll(simplifyTasks, cancellationToken);
```

Comment:
if we change Task.WaitAll(...) to await Task.WhenAll(Tasks).ConfigureAwait(false)

don't we get most of benefit if not all? every tasks given to WhenAll in this particular case has cancellation token (in fact same cancellation token). 

so, What WithCancellation trying to achieve here is not needed. also WithCancellation seems kind of bandage solution where it tries to fix an issue where implementor of tasks didn't do proper cancellation management.

in Roslyn code base, if that is such case, I would say it is just a bug and fix the implementation of that task rather than use WithCancellation.